File: py/practice/square.py
#!/usr/bin/env python3

###############################################################
def make_square_dict(n):
    sd = {}
    for i in range(1, n + 1):
        sd[i] = i * i
    return sd

def make_square_dict_elegant(n):
    return { x : x*x for x in range(1, n + 1) }

def to_string(sd, i):
    if i == 1:
        s = "{}:{}".format(i, sd[i])
    else:
        s = ",{}:{}".format(i, sd[i])
    
    return s

def print_square_dict(sd):
    for i in range(1, len(sd) + 1):
        print(to_string(sd, i), end = "")
    print()

###############################################################
def main():
    n = int(input())
    sd = make_square_dict(n)
    print_square_dict(sd)



###############################################################
if __name__ == "__main__":
    main()

File: cpp/practice/square.cpp
/** 平方运算
 * 
 * 平方是一种运算，sqr(a) = a * a。
 * 比如:
 *      sqr(2) = 2 * 2 = 4
 *      sqr(4) = 4 * 4 = 16
 * 
 * 输入一个较小的正整数 n，请计算它的平方。
 * 然后按照上面例子的格式输出结果。
 * 
 * @arg n, 正整数, n < 100
 * @result 包含运算过程的算式
 * 
 * @test
 * input:
 * 2
 * output:
 * sqr(2) = 2 * 2 = 4
 * 
 * @test
 * input:
 * 4
 * output:
 * sqr(4) = 4 * 4 = 16
 **/

#include <iostream>
#include <map>

/*************************************************************/
std::map<int, int> make_square_dict(int n) {
    std::map<int, int> sd;

    for (int i = 1; i <= n; i++) {
        sd[i] = i * i;
    }

    return sd;
}

std::string to_string(std::map<int, int>& sd, int i) {
    std::string s;

    if (i > 1) {
        s.append(",");
    }

    s.append(std::to_string(i));
    s.append(":");
    s.append(std::to_string(sd[i]));

    return s;
}

void print_square_dict(std::map<int, int>& sd) {
    for (int i = 1; i <= sd.size(); i++) {
        std::cout<< to_string(sd, i);
    }

    std::cout << std::endl;
}

/*************************************************************/
int main(int argc, char* argv[]) {
    std::map<int, int> sd;
    int n;

    std::cin >> n;
    sd = make_square_dict(n);
    print_square_dict(sd);

    return 0;
}
